### PR TITLE
fix(orchestrator): target linked PR canonical issues

### LINKED ISSUE #298: Follow-up: linked PR identifier targeted dispatch canonicalization
## Background

PR #297 review identified a non-blocking corner case introduced by Issue/PR canonical subject dedup. When an operator manually targets a linked PR identifier, for example `run-issue --issue owner/repo#<pr-number>`, canonicalization can remove that PR subject before the identifier filter runs. The result is `dispatched=0` even though the canonical linked Issue may be actionable.

This was reviewed as non-blocking for #278 because the original acceptance criteria focus on project dispatch dedup, standalone PR dispatch, terminal/review suppression, and active-run state sync.

## Scope

- Resolve targeted dispatch identifiers against canonical Issue subjects when the input identifier refers to a linked PR.
- Keep default project scan behavior unchanged: linked PR items should still be deduped under the Issue subject.
- Add regression coverage for `--issue <linked-pr-identifier>` / `run-issue` targeted dispatch.

## Acceptance Criteria

- [ ] Targeting a linked PR identifier dispatches the canonical Issue when that Issue is actionable.
- [ ] Targeting a linked PR identifier does not bypass terminal or `In review` Issue suppression.
- [ ] Targeting a standalone PR identifier still dispatches the standalone PR subject.
- [ ] Existing Issue identifier targeting continues to work.

## Related

- Follow-up from #278 / PR #297 review comments.

### PR DESCRIPTION
**Pull request by moncher-dev**
## Issues

- Fixes #298

## Summary

- Targeted dispatch now matches a canonical Issue when the requested identifier belongs to one of its linked PRs.
- Default project scans still dedupe linked PR project items under the canonical Issue subject.
- The PR branch has been merged with latest `origin/main` after resolving the dispatch service helper conflict.

## Changes

- Added linked PR identifier matching after canonical Issue/PR subject dedup in orchestrator dispatch filtering.
- Added regression coverage for linked PR targeting, terminal and `In review` suppression, standalone PR targeting, existing Issue targeting, and default scan dedup behavior.
- Updated the new targeted dispatch fixtures to include same-repository PR branch metadata required by the current mainline checkout contract.

## Evidence

- `pnpm --filter @gh-symphony/orchestrator test -- dispatch.test.ts` passed (23 tests) after the `origin/main` merge.
- `pnpm lint && pnpm test && pnpm typecheck && pnpm build` passed after the `origin/main` merge.
- Docker E2E blackbox: `docker compose -f docker-compose.e2e.yml -f docker-compose.e2e.events.yml up -d --build`, health check, `happy-path.json` fixture dispatch observed `summary.dispatched: 1` and `activeRuns: 1`; cleaned up with `docker compose ... down` and reset `e2e/fixtures/issues.json` to `[]`.

## Human Validation

- [ ] Confirm `run-issue --issue owner/repo#<linked-pr-number>` dispatches the canonical Issue when actionable.
- [ ] Confirm linked PR targeting does not bypass terminal or `In review` Issue suppression.
- [ ] Confirm standalone PR and direct Issue targeted dispatch still work.

## Risks

- Low. The change only broadens targeted identifier matching inside the `issueIdentifier` dispatch filter; untargeted scan canonicalization remains unchanged.
- The branch now includes a merge from `origin/main`; CI is running on the refreshed head.

### CODE REVIEW

File: packages/orchestrator/src/service.ts [hunk 1]
```diff
@@ -201,6 +201,22 @@ function resolveCanonicalSubjectIssues(
...
+    return true;
+  }
+
+  const linkedPullRequests = Array.isArray(issue.metadata.linkedPullRequests)
+    ? issue.metadata.linkedPullRequests
+    : [];
+  return linkedPullRequests.some(
+    (pullRequest) => pullRequest.identifier === issueIdentifier
+  );
+}
```
Comment (hojinzs) [inline packages/orchestrator/src/service.ts:218]: Non-blocking observation: `TrackedIssueMetadata.linkedPullRequests` is already typed as `TrackedPullRequestContext[] | undefined` in `packages/core/src/contracts/tracker-adapter.ts:52`, so the `Array.isArray(...)` guard here is technically redundant and a `?? []` would read more naturally. That said, the same defensive pattern is used in `resolveCanonicalSubjectIssues` above (and in `packages/core/src/workflow/render.ts`), so keeping it consistent here is defensible — feel free to leave as-is.

One follow-up worth keeping in mind: this matches by `pullRequest.identifier` only, not by `pullRequest.id`. That matches how operators target dispatch today (`owner/repo#N` strings), so it's correct, but if `run-issue` ever starts accepting subject IDs (e.g., `PR_kw...` GraphQL node IDs), this matcher will need to broaden.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/orchestrator/src/dispatch.test.ts [hunk 2]
```diff
@@ -709,6 +709,378 @@ describe("blocker eligibility", () => {
...
+  it.each(["Done", "In review"])(
+    "does not dispatch a canonical Issue in %s when targeting a linked PR identifier",
+    async (state) => {
+      const tempRoot = await mkdtemp(
+        join(tmpdir(), "orchestrator-targeted-pr-suppressed-")
+      );
+      const repository = await createRepositoryFixture(
+        tempRoot,
+        "acme",
+        "platform"
```
Comment (hojinzs) [inline packages/orchestrator/src/dispatch.test.ts:844]: Good coverage — the five tests map 1:1 onto the four acceptance criteria in #298 plus the default-scan dedup invariant. The `it.each(["Done", "In review"])` table for the suppression case is a nice touch and exercises both terminal and review-phase paths.

Small suggestion (optional): consider adding one more case where the target identifier refers to a linked PR but the canonical Issue is **not** present in the project listing (i.e., only the PR appears in `listIssues`). Today that scenario would silently produce `dispatched=0` because `resolveCanonicalSubjectIssues` only iterates surfaced issues. Probably the correct behavior — but pinning it down with a test would document the contract.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by hojinzs — APPROVED** [review]
## Verdict: Approve ✅

### Smoke test
Ran locally against `843cf71`:
- `pnpm --filter @gh-symphony/orchestrator test` → **166/166 passed** (includes the 5 new dispatch cases).
- `pnpm lint` → clean across all packages.
- `pnpm typecheck` → clean.
- `pnpm build` → all packages built.
- CI on the PR head is green (`Test`, `Container Smoke`).

Two failures surfaced in `packages/cli/src/commands/doctor.test.ts` in my sandbox, but they're driven by the absence of the `gh` CLI in this environment and are unrelated to the orchestrator-only changes in this PR (same failures reproduce on `main` here).

### Code review
- **Scope**: 18 added lines in `service.ts` (one new helper, one filter call site swapped) + 407 lines of tests. Minimal blast radius, default-scan path unchanged.
- **Correctness**: `matchesTargetIssueIdentifier` cleanly extends the targeted dispatch filter to recognize linked PR identifiers without touching canonicalization. Suppression of terminal / `In review` Issues is preserved because lifecycle filtering still runs against the canonical Issue's state after the match — confirmed by the parameterized test.
- **Acceptance criteria from #298**: all four are covered by the new tests (linked-PR target dispatches canonical Issue, suppression respected, standalone PR still works, existing Issue targeting unchanged), plus a dedup regression test for the untargeted scan path.
- **Risks**: Low. The change only broadens identifier matching in the `issueIdentifier` branch; no other call sites of `canonicalIssues` are affected.

Two non-blocking inline comments left for the author's consideration (a stylistic note on the `Array.isArray` guard and an optional additional test case). Neither needs to land in this PR.

Note: PR is currently `mergeable_state: dirty` against `main` — main has moved ahead since this branch was cut, so a rebase/merge will be needed before merging, but that's a mechanical step, not a review concern.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by hojinzs — APPROVED** [review]
## Verdict: Approve ✅

### Smoke test (local, head `843cf71`)
- `pnpm --filter @gh-symphony/orchestrator test` → **166/166 passed** (incl. the 5 new `targeted canonical subject dispatch` cases + `keeps default project scans deduped under the canonical Issue subject`).
- `pnpm --filter @gh-symphony/orchestrator test -- dispatch.test.ts` → **23/23 passed**.
- `pnpm lint` → clean across all 11 packages.
- `pnpm typecheck` → clean.
- CI on head: `Prepare` ✅, `Test` ✅, `Container Smoke` ✅ (Agent check still running, unrelated to this PR's code).

### Code review
- **Scope discipline**: helper `matchesTargetIssueIdentifier` (`service.ts:204-218`) is a 14-line additive function with a single call site swap at `service.ts:557-559`. Default-scan path through `resolveCanonicalSubjectIssues` is untouched — linked PR items remain deduped under the canonical Issue for untargeted runs.
- **Correctness**: targeted dispatch now matches a canonical Issue when the input identifier is one of its linked PRs, but lifecycle filtering still runs against the canonical Issue's state, so terminal / `In review` suppression cannot be bypassed. The parameterized `it.each(["Done", "In review"])` test pins this down explicitly.
- **Acceptance criteria from #298**: all four covered (linked-PR target → canonical Issue, suppression respected, standalone PR still dispatches, existing Issue targeting unchanged), plus a regression test for the untargeted scan dedup invariant.
- **Coverage**: `service.ts` reports 93.13% lines / 82.91% branches on this PR — the new helper is fully exercised.
- **Risk**: low. The change only broadens identifier matching inside the `issueIdentifier ?` filter branch; no other call sites of `canonicalIssues` are affected.

### Re: prior inline comments
The two non-blocking notes from the earlier review (the `Array.isArray` guard being technically redundant given the typed `linkedPullRequests`, and an optional test for the "canonical Issue not surfaced in `listIssues`" corner case) still stand but are not blocking — no need to chase them in this PR.

### Operational note
PR is `mergeable_state: dirty` against `main` (main has advanced since the branch was cut). A rebase or merge will be needed before merging — mechanical step, not a review concern.

---
_Reviewed by Claude Code via GitHub Symphony._

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/orchestrator/src/service.ts (hunk 1 above)
Comment (moncher-dev) [reply packages/orchestrator/src/service.ts:218]: Resolved for this cycle: I kept the `Array.isArray` guard because it matches the surrounding defensive metadata handling in `resolveCanonicalSubjectIssues`, and I did not broaden matching to PR node IDs because the current `run-issue --issue` contract is identifier-string based (`owner/repo#N`).

File: packages/orchestrator/src/dispatch.test.ts (hunk 2 above)
Comment (moncher-dev) [reply packages/orchestrator/src/dispatch.test.ts:844]: Triaged for this cycle: I did not add the optional “only PR appears in listIssues” case because #298 is scoped to resolving a linked PR target to the canonical Issue when that canonical Issue is surfaced and actionable. The existing behavior remains that default canonicalization only iterates surfaced subjects; this PR keeps that contract unchanged.
